### DISCUSSION TITLE
/settings endpoint not protected by auth

### DISCUSSION THREAD

**Discussion by vkumar85** [argoproj/argo-cd#9758] [Q&A]
We have been reported by a vulnerability scan, we see that /settings page/endpoint is not protected by any authentication mechanism. Even though, if you try to access any of the menu options there redirects you to authentication, this is still being flagged as vulnerability.

Is there a way to configure to have authentication enabled to access even settings page?

**Comment by crenshaw-dev** [comment]
There's currently no way to disable the endpoint completely without auth, though I'd welcome a PR to add that feature!

**Comment by mabusaa** [comment]
what is the reason behind having /api/v1/settings endpoint without auth ?

**Comment by robinlieb** [comment]
I have created an issue for that topic #16675 and also opened a PR for that.
